light/odr_util.go

```diff
@@ -227,3 +228,18 @@ func GetBloomBits(ctx context.Context, odr OdrBackend, bitIdx uint, sectionIdxLi
 		return result, nil
 	}
 }
+
+// GetTransaction retrieves a canonical transaction by hash and also returns its position in the chain
```

Comment:
Why does this dangle in the air? Why isn't this attached to a light chain or something?

Reply:
It is attached to the `OdrBackend` which is attached to the correct chain. Just like direct database read functions where the chain is selected by the supplied chain database.
It wasn't correct though because I forgot to check whether the header is available and canonical. It is added now.